src/stateMachines/maintenance/changePassword.ts

```diff
@@ -24,167 +21,247 @@ import { SecretManagerCreatorFactory } from '../../factories/SecretManagerCreato
+        entry: assign({
+          message: ({ context, event }) => '',
+          errorMessage: ({ context, event }) => ''
+        }),
         invoke: {
-          src: this.refreshMPS.bind(this),
+          src: 'refreshMPS',
+          input: ({ context }) => (context),
           id: 'refresh-mps',
           onDone: {
             target: 'SUCCESS'
           },
           onError: {
             actions: assign({
-              statusMessage: (context, event) => coalesceMessage('at REFRESH_MPS', event.data)
+              errorMessage: ({ context, event }) => coalesceMessage('at REFRESH_MPS', event.error)
             }),
             target: 'FAILED'
           }
         }
       },
+      ERROR: {
+        entry: sendTo('error-machine', { type: 'PARSE' }),
+        invoke: {
+          src: 'error',
+          id: 'error-machine',
+          input: ({ context, event }) => ({
+            message: context.message,
+            clientId: context.clientId
+          }),
+          onError: {
+            actions: assign({ message: ({ event }) => event.error }),
+            target: 'FAILED'
+          },
+          onDone: 'NEXT_STATE'
+        },
+        on: {
+          ONFAILED: {
+            actions: assign({ errorMessage: ({ context, event }) => coalesceMessage(context.errorMessage, event.output) }),
+            target: 'FAILED'
+          }
+        }
+      },
+      NEXT_STATE: {
+        always: [
+          {
+            guard: 'isGeneralSettings',
+            target: 'GET_GENERAL_SETTINGS'
+          }]
+      },
       FAILED: {
-        type: 'final',
-        data: (context) => (doneFail(context.taskName, context.statusMessage))
+        entry: assign({ statusMessage: ({ context, event }) => 'FAILED' }),
+        type: 'final'
       },
       SUCCESS: {
-        type: 'final',
-        data: (context) => (doneSuccess(context.taskName, context.statusMessage))
+        entry: assign({ statusMessage: ({ context, event }) => 'SUCCESS' }),
+        type: 'final'
+      }
+    },
+    output: ({ context }) => {
+      if (context.statusMessage === 'SUCCESS') {
+        return doneSuccess(context.taskName)
+      } else {
+        return doneFail(context.taskName, context.errorMessage)
       }
     }
   })
 
-  async getGeneralSettings (context: ChangePasswordContext): Promise<AMT.Models.GeneralSettings> {
-    const wsmanXml = amt.GeneralSettings.Get()
-    const rsp = await invokeWsmanCall<AMT.Models.GeneralSettingsResponse>(context.clientId, wsmanXml, 2)
-    const settings = rsp.AMT_GeneralSettings
-    if (!settings) {
-      throw new Error(`invalid response: ${JSON.stringify(rsp)}`)
-    }
-    if (settings.DigestRealm == null || !isDigestRealmValid(settings.DigestRealm)) {
-      throw new Error(`invalid DigestRealm ${rsp.AMT_GeneralSettings?.DigestRealm}`)
-    }
-    logger.debug(`AMT_GeneralSettings: ${JSON.stringify(settings)}`)
-    return settings
+  async getGeneralSettings ({ input }): Promise<any> {
```

Comment:
Can we figure out how to get input typed so it works in xstate?  why did we remove the type for the returned Promise?